src/app/Main/Body/QueryPage/Results/Content-Results.component.tsx

```diff
@@ -1,126 +1,125 @@
-    const { data, setData } = useContext(DataContext);
+    const { data } = useContext(DataContext);
+    const history = useHistory();
+    const { t } = useTranslation(["QueryPage"]);
+
     const rowAttribute: FilterType = table.row;
     const colAttribute: FilterType = table.column;
-
-    const { t } = useTranslation(["QueryPage"]);
+    const isCol: boolean = colAttribute !== "";
+    const isRow: boolean = rowAttribute !== "";
+    const state = { isCol, isRow };
 
     const getTableContext = async (
-        dataToCountIsolates: DbCollection
+        isolateCountProp: IsolateCountedDTO
     ): Promise<void> => {
-        const rowValues = getValuesOfOneFilterAttribute(
-            rowAttribute,
-            data.uniqueValues,
-            filter.selectedFilter
-        );
-        const colValues = getValuesOfOneFilterAttribute(
-            colAttribute,
-            data.uniqueValues,
-            filter.selectedFilter
-        );
-        const nrOfSelectedIsolates = dataToCountIsolates.length;
-        if (rowAttribute.length !== 0 && colAttribute.length !== 0) {
-            const rowsWithIsolates = generateRowsWithIsolates(
-                dataToCountIsolates,
-                colAttribute,
-                rowAttribute,
-                rowValues,
-                colValues,
-                "both"
-            );
-            const rowsWithPercent = calculateRowsWithPercent(rowsWithIsolates, nrOfSelectedIsolates)
-            setState({
-                isCol: true,
-                isRow: true,
-            });
-            setColumnAttributes(colValues);
+        if (!isCol && !isRow) {
             setTable({
                 ...table,
-                statisticDataAbsolute: rowsWithIsolates,
-                statisticDataPercent: rowsWithPercent,
+                statisticDataAbsolute: [],
             });
-        } else if (rowAttribute.length === 0 && colAttribute.length !== 0) {
-            const rowsWithIsolates = generateRowsWithIsolates(
-                dataToCountIsolates,
-                colAttribute,
-                rowAttribute,
-                [t("Results.TableHead")],
-                colValues,
-                "col"
-            );
-            const rowsWithPercent = calculateRowsWithPercent(rowsWithIsolates, nrOfSelectedIsolates)
-            setState({
-                isCol: true,
-                isRow: false,
+        } else {
+            const rowValues: string[] = !isRow
+                ? [t("Results.TableHead")]
+                : (_.isEmpty(filter.selectedFilter[rowAttribute])
+                ? data.uniqueValues[rowAttribute]
+                : filter.selectedFilter[rowAttribute])
+
+            const colValues: string[] = !isCol
+                ? [t("Results.TableHead")]
+                : (_.isEmpty(filter.selectedFilter[colAttribute])
+                ? data.uniqueValues[colAttribute]
+                : filter.selectedFilter[colAttribute])

```

Comment:
try to avoid nested ? : operators. They should only be used for short and simple statements. So for readability i would prefer something like:
rowValues = t("...")
if(isRow) {
const isEmpty = _.isEmpty(...)
rowValues = isEmpty ? data.uniqueValues : filter.selectedValues
}

Reply:
Done. Commit "Make it simple"